django/db/backends/postgresql/operations.py

```diff
@@ -368,3 +385,15 @@ def on_conflict_suffix_sql(self, fields, on_conflict, update_fields, unique_fiel
             update_fields,
             unique_fields,
         )
+
+
+if is_psycopg3:
+    from psycopg import ClientCursor
+
+    def compose(query, params, connection):
```

Comment:
Is there a better place to put this? Since we `connection` is a required argument we could put it on the connection object directly, would save some imports :)

Reply:
Moved to the `DatabaseOperations`. I also added `mogrify` hook in `psycopg_any` to make it easier to maintain (per [Simon's comment](https://github.com/django/django/pull/15687#pullrequestreview-1167108292)).